community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/staging/OnDemandDetailsExecutionMonitor.java

```diff
@@ -0,0 +1,279 @@
+        this.actions.put( "gc", Pair.of( "Trigger GC", this::triggerGC ) );
+        this.gcMonitor = new MeasureDoNothing( "Importer GC monitor", gcBlockTime, 100, 200 );
+    }
+
+    @Override
+    public void initialize( DependencyResolver dependencyResolver )
+    {
+        out.println( "Interactive command list (end with ENTER):" );
+        actions.forEach( ( key, action ) -> out.println( "  " + key + ": " + action.first() ) );
+        out.println();
+        gcMonitor.start();
+    }
+
+    @Override
+    public void start( StageExecution execution )
+    {
+        details.add( current = new StageDetails( execution, gcBlockTime ) );
+    }
+
+    @Override
+    public void end( StageExecution execution, long totalTimeMillis )
+    {
+        current.collect();
+    }
+
+    @Override
+    public void done( long totalTimeMillis, String additionalInformation )
+    {
+        if ( printDetailsOnDone )
+        {
+            printDetails();
+        }
+        gcMonitor.stopMeasuring();
+    }
+
+    @Override
+    public long nextCheckTime()
+    {
+        return currentTimeMillis() + 500;
+    }
+
+    @Override
+    public void check( StageExecution execution )
+    {
+        current.collect();
+        reactToUserInput();
+    }
+
+    private void printDetails()
+    {
+        printDetailsHeadline();
+        long totalTime = 0;
+        for ( StageDetails stageDetails : details )
+        {
+            stageDetails.print( out );
+            totalTime += stageDetails.totalTimeMillis;
+        }
+
+        printIndented( out, "Environment information:" );
+        printIndented( out, "  Free physical memory: " + bytes( OsBeanUtil.getFreePhysicalMemory() ) );
+        printIndented( out, "  Max VM memory: " + bytes( Runtime.getRuntime().maxMemory() ) );
+        printIndented( out, "  Free VM memory: " + bytes( Runtime.getRuntime().freeMemory() ) );
+        printIndented( out, "  GC block time: " + duration( gcBlockTime.getGcBlockTime() ) );
+        printIndented( out, "  Duration: " + duration( totalTime ) );
+        out.println();
+    }
+
+    private void printDetailsHeadline()
+    {
+        out.println();
+        out.println();
+        printIndented( out, "******** DETAILS " + date() + " ********" );
+        out.println();
+
+        // Make sure that if user is interested in details then also print the entire details set when import is done
+        printDetailsOnDone = true;
+        monitor.detailsPrinted();
+    }
+
+    private void printDetailsForCurrentStage()
+    {
+        printDetailsHeadline();
+        if ( !details.isEmpty() )
+        {
+            details.get( details.size() - 1 ).print( out );
+        }
+    }
+
+    private static void printIndented( PrintStream out, String string )
+    {
+        out.println( "\t" + string );
+    }
+
+    private void triggerGC()
```

Comment:
this is useless action right since we disable explicit GC everywhere?

Reply:
`disable explicit GC everywhere` is that a strict policy implemented somewhere, if so where? With that said it has been useful while testing and may be useful for others, but it's likely not for most cases.